fuel-tx/src/transaction/fee.rs

```diff
@@ -180,15 +173,18 @@ pub trait Chargeable: field::Inputs + field::Witnesses + field::Policies {
         gas_costs: &GasCosts,
         fee: &FeeParameters,
         used_gas: Word,
+        gas_price: u64,
     ) -> Option<Word> {
         // We've already charged the user for witnesses as part of the minimal gas and all
         // execution required to validate transaction validity rules.
         let min_gas = self.min_gas(gas_costs, fee);
 
         let total_used_gas = min_gas.saturating_add(used_gas);
-        let used_fee = gas_to_fee(total_used_gas, self.price(), fee.gas_price_factor);
+        let used_fee = gas_to_fee(total_used_gas, self.price(), gas_price);
```

Comment:
```suggestion
        let used_fee = gas_to_fee(total_used_gas, gas_price, fee.gas_price_factor);
```

The gas_price_factor is a global consensus constant param, and shouldn't be removed.